File: apina-core/src/test/java/fi/evident/apina/utils/PatternSetTest.java
package fi.evident.apina.utils;

import org.junit.Test;

import static org.hamcrest.Matchers.is;
import static org.junit.Assert.assertThat;

public class PatternSetTest {

    private final PatternSet set = new PatternSet();

    @Test
    public void emptySetMatchesNothing() {
        assertThat(set.test(""), is(false));
        assertThat(set.test("foo"), is(false));
    }

    @Test
    public void setWithSinglePatternMatchesIfPatternMatches() {
        set.addPattern("foo.*");

        assertThat(set.test(""), is(false));
        assertThat(set.test("bar"), is(false));
        assertThat(set.test("foo"), is(true));
        assertThat(set.test("foobar"), is(true));
    }

    @Test
    public void setWithMultiplePatternsMatchesIfAnyPatternMatches() {
        set.addPattern("foo.*");
        set.addPattern("bar.*");
        set.addPattern("baz.*");

        assertThat(set.test(""), is(false));

        assertThat(set.test("foo"), is(true));
        assertThat(set.test("bar"), is(true));
        assertThat(set.test("baz"), is(true));
        assertThat(set.test("baz-quux"), is(true));

        assertThat(set.test("quux"), is(false));
    }
}

File: apina-core/src/test/kotlin/fi/evident/apina/utils/PatternSetTest.kt
package fi.evident.apina.utils

import org.junit.jupiter.api.Test
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class PatternSetTest {

    private val set = PatternSet()

    @Test
    fun `empty set matches nothing`() {
        assertFalse("" in set)
        assertFalse("foo" in set)
    }

    @Test
    fun `set with single pattern matches if pattern matches`() {
        set.addPattern("foo.*")

        assertFalse("" in set)
        assertFalse("bar" in set)
        assertTrue("foo" in set)
        assertTrue("foobar" in set)
    }

    @Test
    fun `set with multiple patterns matches if any pattern matches`() {
        set.addPattern("foo.*")
        set.addPattern("bar.*")
        set.addPattern("baz.*")

        assertFalse("" in set)

        assertTrue("foo" in set)
        assertTrue("bar" in set)
        assertTrue("baz" in set)
        assertTrue("baz-quux" in set)

        assertFalse("quux" in set)
    }
}
